datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -280,25 +280,64 @@ where
         .collect()
 }
 
+fn arith_decimal_scalar<F>(
```

Comment:
It would be great to move some / all of this decimal manipulation into the arrow-rs crate at some point

Reply:
thanks for your reminder and I have a plan to do this.

Reply:
But now I find a bug in the arithmetic operation. https://github.com/apache/arrow-datafusion/blob/c91efc27658e58264c4f346a5cfdec8810179e90/datafusion/physical-expr/src/expressions/binary.rs#L302 and https://github.com/apache/arrow-datafusion/blob/c91efc27658e58264c4f346a5cfdec8810179e90/datafusion/physical-expr/src/expressions/binary.rs#L307

In the divide method, we may lose of precision by converting the `i128` to `f64`.
We use the `f64` to store the larger range of data, but we lose the precision.

I will try to figure out this issue and resolve this.